tests/std/tests/GH_000625_vector_bool_optimization/test.cpp

```diff
@@ -213,8 +215,48 @@ bool test_count() {
     return true;
 }
 
+// Also test the behavior of a huge vector<bool, A> whose size is greater than SIZE_MAX, which is pratical on 32-bit
+// platforms.
+template <class T>
+struct huge_allocator {
+    huge_allocator() = default;
+    template <class U>
+    constexpr huge_allocator(const huge_allocator<U>) noexcept {}
+
+    using value_type      = T;
+    using size_type       = uint64_t;
+    using difference_type = int64_t;
+
+    T* allocate(uint64_t n) {
+        return allocator<T>{}.allocate(static_cast<size_t>(n));
+    }
+
+    void deallocate(T* p, uint64_t n) {
+        allocator<T>{}.deallocate(p, static_cast<size_t>(n));
+    }
+};
+
+void test_huge_vector_bool() {
+    constexpr uint64_t small_bit_length = 0x7000'4321ULL;
+    constexpr uint64_t large_bit_length = 0x1'2000'4321ULL; // overflows uint32_t

```

Comment:
I am somewhat concerned about the test reliability implications of consuming 604 MB at runtime. That said, this is less than the compiler memory consumption of certain ranges tests, so I have no concrete objection to this.